File: backend/app/schemas/job.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime


class JobResponse(BaseModel):
    id: UUID
    video_id: UUID
    video_url: str | None = None
    status: str
    comments_scraped: int
    error_message: str | None = None
    started_at: datetime
    finished_at: datetime | None = None
    created_at: datetime | None = None

    model_config = {"from_attributes": True}


class JobListResponse(BaseModel):
    data: list[JobResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class ScrapeStartResponse(BaseModel):
    job_id: UUID
    status: str

File: frontend/src/types/job.ts
export type JobStatus = "PENDING" | "RUNNING" | "SUCCESS" | "FAILED";

export interface ScrapeJob {
  id: string;
  video_id: string;
  video_url: string;
  status: JobStatus;
  target_comments: number;
  comments_scraped: number;
  error_message: string | null;
  started_at: string;
  finished_at: string | null;
  created_at: string;
}

export interface JobListResponse {
  data: ScrapeJob[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface JobListParams {
  page?: number;
  page_size?: number;
  status?: JobStatus;
  sort_by?: string;
  sort_order?: "asc" | "desc";
}
